inc/mlas.h

```diff
@@ -0,0 +1,1699 @@
+/*++
+
+Copyright (c) Microsoft Corporation. All rights reserved.
+
+Licensed under the MIT License.
+
+Module Name:
+
+    mlas.h
+
+Abstract:
+
+    This module contains the public data structures and procedure prototypes
+    for the Microsoft Machine Learning algebra subprogram library.
+
+--*/
+
+#pragma once
+
+#include <cstddef>
+#include <cstdlib>
+#include <cstdint>
+#include <functional>
+//
+// Define the calling convention for Windows targets.
+//
+
+#if (_MSC_VER >= 800) || defined(_STDCALL_SUPPORTED)
+#define MLASCALL __stdcall
+#else
+#define MLASCALL
+#endif
+
+//
+// Define the target architecture.
+//
+
+#if (defined(_M_AMD64) && !defined(_M_ARM64EC)) || defined(__x86_64__)
+#define MLAS_TARGET_AMD64
+#endif
+#if defined(_M_IX86) || defined(__i386__)
+#define MLAS_TARGET_IX86
+#endif
+#if defined(MLAS_TARGET_AMD64) || defined(MLAS_TARGET_IX86)
+#define MLAS_TARGET_AMD64_IX86
+#endif
+#if defined(_M_ARM64) || defined(__aarch64__)
+#define MLAS_TARGET_ARM64
+#endif
+#if defined(_M_ARM64EC)
+#define MLAS_TARGET_ARM64EC
+#endif
+#if defined(_M_ARM) || defined(__arm__)
+#define MLAS_TARGET_ARM
+#endif
+#if defined(MLAS_TARGET_ARM64) || defined(MLAS_TARGET_ARM64EC) || defined(MLAS_TARGET_ARM)
+#define MLAS_TARGET_ARM_ANY
+#endif
+
+#if defined(__VSX__)
+#define MLAS_TARGET_POWER
+#endif
+#if defined(__wasm__)
+#define MLAS_TARGET_WASM
+#if defined(__wasm_simd128__)
+#define MLAS_TARGET_WASM_SIMD
+#else
+#define MLAS_TARGET_WASM_SCALAR
+#endif
+#endif
+
+//
+// Define the support levels for the target architecture.
+//
+
+#if defined(MLAS_TARGET_AMD64) || defined (MLAS_TARGET_POWER)
+#define MLAS_SUPPORTS_GEMM_DOUBLE
+#endif
+
+#if (!defined(_MSC_VER)) || (_MSC_VER >= 1930)
+#if defined(MLAS_TARGET_ARM64) || defined(MLAS_TARGET_ARM64EC)
+#if !defined(__APPLE__)
+// Had to temporary disable fp16 under APPLE ARM64, as compiling
+// the source files require a hardware specific compilation flag.
+// When building an universial binary for APPLE, this flag would
+// cause trouble for x64 target.
+
+#define MLAS_F16VEC_INTRINSICS_SUPPORTED
+
+#endif // 
+#endif // ARM64
+#endif // Visual Studio 16 or earlier does not support fp16 intrinsic
+
+//
+// Basic Linear Algebra Subprograms (BLAS) types.
+//
+
+#ifndef CBLAS_ENUM_DEFINED_H
+#define CBLAS_ENUM_DEFINED_H
+typedef enum { CblasNoTrans=111, CblasTrans=112, CblasConjTrans=113 } CBLAS_TRANSPOSE;
+typedef enum { CblasUpper=121, CblasLower=122 } CBLAS_UPLO;
+typedef enum { CblasNonUnit=131, CblasUnit=132 } CBLAS_DIAG;
+typedef enum { CblasLeft=141, CblasRight=142} CBLAS_SIDE;
+#endif
+
+//
+// Forward declare the thread pool implementation class and half precision floating point.
+//
+// N.B. Avoid including ONNX Runtime headers here to keep the dependencies for
+// standalone MLAS test executables smaller.
+//
+
+namespace onnxruntime {
+    struct MLFloat16;
+};  // namespace onnxruntime
+
+namespace ov {
```

Comment:
MLAS library (even if this is our fork) should  know nothing about OpenVINO. So I would propose to:
1. Avoid usage of ov::cpu namespace.
2. Introduce `IMlasThreadPool` interface with 3 **pure** methods: `DegreeOfParallelism`, `TrySimpleParallelFor` and `getCacheSize`. Then on CPU plugin level you will implement this interface using already existed `OVThreadPool`